src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8753,31 +8706,28 @@ private TypeWithAnnotations GetDeclaredParameterResult(ParameterSymbol parameter
 
         public override BoundNode? VisitImplicitIndexerAccess(BoundImplicitIndexerAccess node)
         {
-            // The argument will be visited as part of VisitImplicitIndexerValuePlaceholder (for the first argument)
-            // to maintain proper order of evaluation
-            AddPlaceholder(node.ArgumentPlaceholders[0], node.Argument);
+            VisitRvalue(node.Receiver);
+            var receiverResult = _visitResult;
+            VisitRvalue(node.Argument);
+
+            EnsurePlaceholdersToResultMap();
+            _resultForPlaceholdersOpt.Add(node.ReceiverPlaceholder, (node.Receiver, receiverResult));
             VisitRvalue(node.IndexerOrSliceAccess);
-            RemovePlaceholder(node.ArgumentPlaceholders[0]);
+            _resultForPlaceholdersOpt.Remove(node.ReceiverPlaceholder);
```

Comment:
Not blocking this PR, since it's not the main purpose, but consider adding the visit for LengthAccess too, now that we can.

Reply:
> Not blocking this PR, since it's not the main purpose, but consider adding the visit for LengthAccess too, now that we can.

I am not sure if that would be the right thing to do. The length is not always evaluated and I prefer to not deal with the logic to figure that out here. The receiver is checked already, there is pretty much nothing else we can learn.